website/docs/concepts/provider_lifecycles.mdx

```diff
@@ -0,0 +1,77 @@
+---
+title: Provider Lifecycles
+---
+
+## When does my Provider get created and disposed?
+
+The states that all different types of providers can go through are the same:
+
+- Disposed / Uninitialized
+- Creating
+- Alive
+
+### Disposed / Uninitialized
+
+An uninitialized provider does not take up any memory since its state is not initialized.
+Essentially it is just a definition of how to create state when a widget needs it.
+It will stay that way until an Alive provider or the UI reads or watches it.
+
+### Creating -> Alive
+
+When an uninitialized provider is being read or watched it will be created.
+
+When a provider is created your provider creation function will be run. Any
+providers that you read or watch will be created as needed and their state will be retrieved.
+
+If there are any circular dependencies during this creation process Riverpod will throw an error.
+The best way to fix this error is to redesign your dependencies to have a uni-directional dataflow.
+
+The provider's state is stored in a [ProviderContainer]. In a Flutter app this container is in a [ProviderScope] widget.
+
+### Alive
+
+When your provider is Alive, changes to its state will cause dependent providers and/or the dependent UI to rebuild.
```

Comment:
More importantly, the provider will also automatically re-initialize if one of its dependencies has changed.

Reply:
That captured via this statement, but is from the perspective of the dependent provider. I've added a statement that explains it from the other perspective.